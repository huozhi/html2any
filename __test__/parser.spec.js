const html1 = require('./mock/html1')
const html2 = require('./mock/html2')
const tokenizer = require('../src/tokenizer')
const parser = require('../src/parser')

const logJSON = x => console.log(JSON.stringify(x, null, 2))

it('parse correctly', () => {
  const tks = tokenizer(html1)
  expect(parser(tks)).toMatchSnapshot()
})

it('parse correctly', () => {
  const tks = tokenizer(html2)
  expect(parser(tks)).toMatchSnapshot()
})
