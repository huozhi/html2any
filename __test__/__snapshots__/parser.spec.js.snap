// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parse correctly 1`] = `
Array [
  Object {
    "attribues": Object {},
    "children": Array [
      Object {
        "attribues": Object {},
        "children": undefined,
        "name": "hr",
      },
      Object {
        "attribues": Object {
          "data-index": "2",
          "src": "a.jpg",
        },
        "children": undefined,
        "name": "img",
      },
      Object {
        "attribues": Object {
          "id": "x3",
        },
        "children": Array [
          "1234 &amp;",
        ],
        "name": "p",
      },
      Object {
        "attribues": Object {
          "class": "d3 d2\\" i",
        },
        "children": Array [
          Object {
            "attribues": Object {
              "disabled": true,
              "value": "123",
            },
            "children": undefined,
            "name": "input",
          },
        ],
        "name": "div",
      },
    ],
    "name": "div",
  },
]
`;

exports[`parse correctly 2`] = `
Array [
  Object {
    "attribues": Object {},
    "children": Array [
      Object {
        "attribues": Object {},
        "children": Array [
          Object {
            "attribues": Object {},
            "children": undefined,
            "name": "br",
          },
        ],
        "name": "p",
      },
      Object {
        "attribues": Object {
          "style": "text-align: center; font-size: 15px;\\" data-mce-styl",
        },
        "children": Array [
          Object {
            "attribues": Object {
              "alt": "TinyMCE",
              "height": "97\\" data-mce-sr",
              "src": "https://www.tinymce.com/images/glyph-tinymce@2x.png",
              "width": "110",
            },
            "children": undefined,
            "name": "img",
          },
        ],
        "name": "p",
      },
      Object {
        "attribues": Object {
          "style": "text-align: center; color: #666; font-size: 15px; font-family: avenir; font-weight: 200;\\" data-mce-styl",
        },
        "children": Array [
          "This demo includes these TinyMCE Premium",
          Object {
            "attribues": Object {
              "class": "mce-spellchecker-word",
              "data-mce-bogus": "1",
              "data-mce-word": "Plugins\\" data-mce-inde",
            },
            "children": Array [
              "Plugins",
            ],
            "name": "span",
          },
          ":",
          Object {
            "attribues": Object {
              "data-mce-href": "powerpaste\\" data-mce-linkchecker-statu",
              "href": "https://www.tinymce.com/powerpaste",
            },
            "children": Array [
              "PowerPaste",
            ],
            "name": "a",
          },
          "(improved Word paste),",
          Object {
            "attribues": Object {
              "data-mce-href": "accessibility-checker\\" data-mce-linkchecker-statu",
              "href": "https://www.tinymce.com/accessibility-checker",
            },
            "children": Array [
              "Accessibility Checker",
            ],
            "name": "a",
          },
          ",",
          Object {
            "attribues": Object {
              "data-mce-href": "spell-checker-pro\\"
      data-mce-linkchecker-statu",
              "href": "https://www.tinymce.com/spell-checker-pro",
            },
            "children": Array [
              "Spell Checker Pro",
            ],
            "name": "a",
          },
          "&amp; file management with",
          Object {
            "attribues": Object {
              "data-mce-href": "http://www.moxiemanager.com\\" data-mce-linkchecker-statu",
              "href": "http://www.moxiemanager.com",
            },
            "children": Array [
              "MoxieManager",
            ],
            "name": "a",
          },
          ".",
        ],
        "name": "p",
      },
      Object {
        "attribues": Object {
          "style": "text-align: center; color: #666; font-size: 15px; font-family: avenir; font-weight: 200;\\"
    data-mce-styl",
        },
        "children": Array [
          "TinyMCE is a platform indepen",
          Object {
            "attribues": Object {},
            "children": Array [
              "dent web-",
            ],
            "name": "strong",
          },
          "based JavaScript HTML WYSIWYG",
          Object {
            "attribues": Object {},
            "children": undefined,
            "name": "br",
          },
          "editor control released as open source",
          Object {
            "attribues": Object {},
            "children": Array [
              "under",
            ],
            "name": "em",
          },
          "LGPL.",
        ],
        "name": "p",
      },
      Object {
        "attribues": Object {
          "style": "text-align: center; color: #666; font-size: 15px; font-family: avenir; font-weight: 200;\\" data-mce-styl",
        },
        "children": Array [
          Object {
            "attribues": Object {},
            "children": Array [
              "TinyMCE enables you to convert HTML",
              Object {
                "attribues": Object {
                  "class": "mce-spellchecker-word",
                  "data-mce-bogus": "1",
                  "data-mce-word": "textarea\\" data-mce-inde",
                },
                "children": Array [
                  "textarea",
                ],
                "name": "span",
              },
              "fields or other HTML elements to editor instances.",
            ],
            "name": "em",
          },
        ],
        "name": "p",
      },
      Object {
        "attribues": Object {},
        "children": Array [
          Object {
            "attribues": Object {},
            "children": undefined,
            "name": "br",
          },
        ],
        "name": "p",
      },
    ],
    "name": "div",
  },
]
`;
